### PR TITLE
Custom environment variables in the build hook.

### PR DESCRIPTION
**Pull request by matthiaz**
Adding information on how to use custom environment variables in the build hook.

### CODE REVIEW

**Comment by Crell** [comment]
This is way too complicated for people to do.  If we want people to use jq, we should just pre-install it.  (We've discussed that before.)  If they want to access an individual variable directly, they should prefix it with `env:` so it's available as a bare variable and remove the need to do any decoding at all.

Frankly PLATFORM_VARIABLES is kind of vestigial at this point.

File: src/configuration/app/build.md [hunk 1]
```diff
@@ -163,3 +163,32 @@ hooks:
         fi
         drush -y updatedb
 ```
+
+## How can I use my custom environment variables?
+
+Custom environment variables can be defined in the console. They will be available through the $PLATFORM_VARIABLES but to extract them, you need a couple of extra steps.
```
Comment (chadwcarlson) [inline src/configuration/app/build.md]: Custom environment variables can be defined using the [management console](/administration/web/configure-project.md#variables) or with the [Platform.sh CLI](/development/variables.md). They will be available within your during build as the base64-encoded JSON `PLATFORM_VARIABLES`. 

As an example, you can unpack `PLATFORM_VARIABLES` with a tool like `jq` in your build hook to access them:

```yaml
```

**Comment by matthiaz** [comment]
> This is way too complicated for people to do. If we want people to use jq, we should just pre-install it. (We've discussed that before.) If they want to access an individual variable directly, they should prefix it with `env:` so it's available as a bare variable and remove the need to do any decoding at all.
> 
> Frankly PLATFORM_VARIABLES is kind of vestigial at this point.

I agree, jq should be added. But before we do, we still have to give the customer something to work with. (This documentation comes straight out of communication to certain customers)

**Comment by Crell** [comment]
The answer for the 99.9999% case is "use an env: prefix on your variable and none of this is needed".  That's the answer we should be pointing people to, especially on the shell.

File: src/configuration/app/build.md (hunk 1 above)
Comment (matthiaz) [reply src/configuration/app/build.md]: Err... I like your text better than mine. How do I accept it?

**Comment by Crell** [comment]
It's a totally different approach so probably just a new PR.  Or we do nothing, because the env: prefix is already documented.

File: src/configuration/app/build.md (hunk 1 above)
Comment (chadwcarlson) [reply src/configuration/app/build.md]: Bad reviewing here on my part. Second sentence should be:

> They will be available during build as the base64-encoded JSON `PLATFORM_VARIABLES`.

Comment (chadwcarlson) [reply src/configuration/app/build.md]: As far as I know use what you like of my edit, push it, then I approve the changes. Not sure if there's an "accept changes".

**Comment by matthiaz** [comment]
@Crell is right. it is already mentioned in the documentation to use the prefix. So ignore this PR, I'll do a better job at RTFM :-)

**Pull request closed without merging**